main/uart_nic.cpp

```diff
@@ -738,9 +752,16 @@ static void IRAM_ATTR uart0_tx_task(void *arg) {
         struct uart0_tx_queue_item queue_item;
         if (xQueueReceive(uart0_tx_queue, &queue_item, portMAX_DELAY) == pdTRUE) {
 
-            // send fix-sized part of the message (intron + header)
+            // send fix-sized part of the message (intron + header + checksum)
             tx_message.header = queue_item.header;
-            uart0_tx_bytes(tx_message.bytes, sizeof(tx_message.bytes));
+            uint32_t crc = 0;
+            crc = crc32_le(crc, tx_message.intron.data(), tx_message.intron.size());
+            crc = crc32_le(crc, reinterpret_cast<uint8_t *>(&tx_message.header), sizeof(tx_message.header));
```

Comment:
Might be nice if you write a helper for CRC that accepts `std::span<uint8_t>`. Lots of reinterpret_casting could go away.

Reply:
I don't think it would, since I would need to `reinterpret_cast` the objects to make the `std::span`s anyway or would need to have an template helper that would automatically do that.